mvrx/src/main/kotlin/com/airbnb/mvrx/PersistState.kt

```diff
@@ -47,6 +47,16 @@ internal fun <T : MvRxState> T.persistState(validation: Boolean = false): Bundle
     return bundle
 }
 
+private fun <T : MvRxState> Class<out T>.primaryConstructor(): Constructor<*>? {
```

Comment:
As a further optimization, instead of iterating through the primary constructor twice, we could surround the loop above with the constructors loop then break it once a constructor with the annotation is found